core/invoker/src/main/scala/whisk/core/invoker/InvokerReactive.scala

```diff
@@ -156,7 +166,14 @@ class InvokerReactive(
                     Future.failed(new IllegalStateException())
             }
         }.recover {
-            case _ =>
+            case t =>
+                // If the action cannot be found, the user has concurrently deleted it,
+                // making this an application error. All other errors are considered system
+                // errors and should cause the invoker to be considered unhealthy.
+                val response = t match {
+                    case _: NoDocumentException => ActivationResponse.applicationError("action could not be found")
+                    case _                      => ActivationResponse.whiskError("action could not be found")
```

Comment:
Since these are user visible messages, maybe put them in the Messages singleton? I started some time ago to consolidate all messages there which in the future we might want to internationalize. (Is there a test of this case?)